pkg/providers/tinkerbell/assert.go

```diff
@@ -98,6 +98,27 @@ func NewIPNotInUseAssertion(client networkutils.NetClient) ClusterSpecAssertion
 	}
 }
 
+// AssertTinkerbellIPNotInUse ensures tinkerbell ip isn't in use
+func AssertTinkerbellIPNotInUse(client networkutils.NetClient) ClusterSpecAssertion {
+	return func(spec *ClusterSpec) error {
+		ip := spec.DatacenterConfig.Spec.TinkerbellIP
+		if err := validateIPUnused(client, ip); err != nil {
+			return fmt.Errorf("tinkerbellIP <%s> is already in use, please provide a unique IP", ip)
+		}
+		return nil
+	}
+}
+
+// AssertTinkerbellIPAndControlPlaneIPNotSame ensures tinkerbell ip and controlplane ip are not the same
+func AssertTinkerbellIPAndControlPlaneIPNotSame(spec *ClusterSpec) error {
+	tinkerbellIP := spec.DatacenterConfig.Spec.TinkerbellIP
+	controlPlaneIP := spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host
+	if err := validateIPNotSame(tinkerbellIP, controlPlaneIP); err != nil {
```

Comment:
instead of calling this function `validateIPNotSame`, I would just compare the values here directly
```suggestion
	if tinkerbellIP == controlPlaneIP {
```